src/StarcounterClientFiles/service-worker.tmpl.js

```diff
@@ -0,0 +1,111 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+// Names of the two caches used in this version of the service worker.
+// Change to v2, etc. when you update any of the local resources, which will
+// in turn trigger the install event again.
+const PRECACHE = REPLACE_ME_WITH_WITH_PRE_CACHED_FILES_HASH;
+const RUNTIME = 'runtime';
+
+// A list of local resources we always want to be cached.
+const PRECACHE_WITHOUT_SYS = REPLACE_ME_WITH_WITH_PRE_CACHE_URLS;
+
+/* there is a couple hundred URLs all prefixed with `sys`. Removing those `sys`es and adding them here saves ~1KB */
+const PRECACHE_URLS = PRECACHE_WITHOUT_SYS.map(url => `sys/${url}`);
```

Comment:
This microoptimisation makes by brain melt. What's the point of saving 1 KB of client side storage if it makes the code incomprehensible? Please remove that if possible.